examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+
+init_params = strong_ent_layers_uniform(num_layers, num_wires)
+
+# Optimizing using exact gradient descent
+
+cost_GD = []
+params_GD = init_params
+opt = qml.GradientDescentOptimizer(eta)
+
+for _ in range(steps):
+    cost_GD.append(qnode_GD(params_GD))
+    params_GD = opt.step(qnode_GD, params_GD)
+
+# Optimizing using stochastic gradient descent with shots=1
+
+dev_SGD.shots = 1
+cost_SGD1 = []
+params_SGD1 = init_params
+opt = qml.GradientDescentOptimizer(eta)
+
+for _ in range(steps):
+    cost_SGD1.append(qnode_SGD(params_SGD1))
+    params_SGD1 = opt.step(qnode_SGD, params_SGD1)
+
+# Optimizing using stochastic gradient descent with shots=100
+
+dev_SGD.shots = 100
+cost_SGD100 = []
+params_SGD100 = init_params
+opt = qml.GradientDescentOptimizer(eta)
+
+for _ in range(steps):
+    cost_SGD100.append(qnode_SGD(params_SGD100))
+    params_SGD100 = opt.step(qnode_SGD, params_SGD100)
+
+
+##############################################################################
+# We can now plot the cost against optimization step for the three cases.
+
+from matplotlib import pyplot as plt
+
+plt.style.use("seaborn")
+plt.plot(cost_GD[:100], label="Vanilla gradient descent")
+plt.plot(cost_SGD100[:100], label="Stochastic gradient descent, $shots=100$")
+plt.plot(cost_SGD1[:100], ".", label="Stochastic gradient descent, $shots=1$")
+
+# analytic ground state
+min_energy = min(np.linalg.eigvalsh(H))
+plt.hlines(min_energy, 0, 100, linestyles=":", label="Ground-state energy")
+
+plt.ylabel("Cost function value")
+plt.xlabel("Optimization steps")
+plt.legend()
+plt.show()
+
+##############################################################################
+# Using the trained parameters from each optimization strategy, we can
+# evaluate the analytic quantum device:
+
+print("Vanilla gradient descent min energy = ", qnode_GD(params_GD))
+print("Stochastic gradient descent (shots=100) min energy = ", qnode_GD(params_SGD100))
+print("Stochastic gradient descent (shots=1) min energy = ", qnode_GD(params_SGD1))
+
+##############################################################################
+# Amazingly, we see that even the ``shots=1`` optimization converged
+# to a reasonably close approximation of ground-state energy!
+
+##############################################################################
+# Doubly stochastic gradient descent for VQE
+# ------------------------------------------
+#
+# As noted in `Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__,
+# variational quantum algorithms often include terms consisting of linear combinations
+# of expectation values. This is true of the parameter-shift rule (where the
+# gradient of each parameter is determined by shifting the parameter by macroscopic
+# amounts and taking the difference), as well as VQE, where the Hamiltonian
+# is usually decomposed into a sum of Pauli expectation values.
+#
+# Consider the Hamiltonian from the previous section. As this Hamiltonian is an

```

Comment:
```suggestion
# Consider the Hamiltonian from the previous section. As this Hamiltonian is a
```

I think in this case the "n" is phonetically supressed...

Reply:
Good catch!